### PR TITLE
Bug 1826140: add infra flag and configure raft for Azure

### PR DESCRIPTION
**Pull request by alaypatel07**
This PR adds the ability to parse the platform for the cluster
during bootstrap. Since the default for the flag is set appropriately
it is expected to work out of the box. The next step would be
explicitly supplying the flag with value in the installer and making it
required.

This also paves the way to check if libvirt and render the etcd CR
with unsupported config override from #279 eliminating the need
for manual run

### CODE REVIEW

File: bindata/bootkube/bootstrap-manifests/etcd-member-pod.yaml [hunk 1]
```diff
@@ -120,6 +120,8 @@ spec:
       #!/bin/sh
       set -euo pipefail
 
+      {{ if eq .Platform "Azure"}} export ETCD_HEARTBEAT_INTERVAL="500" {{ end }}
+
```
Comment (hexfusion) [inline bindata/bootkube/bootstrap-manifests/etcd-member-pod.yaml:124]: we also need to bump leader election timeout[1].

[1] https://github.com/openshift/cluster-etcd-operator/blob/8f7718d45c4a31235c276322365ba10432921adb/pkg/etcdenvvar/etcd_env.go#L225

Comment (alaypatel07) [reply bindata/bootkube/bootstrap-manifests/etcd-member-pod.yaml:124]: updated

**Comment by hexfusion** [comment]
In the future, we may want something more robust to handle additional provider cases but I think this solves our problem. Thanks @alaypatel07

File: pkg/cmd/render/render.go [hunk 2]
```diff
@@ -154,6 +158,11 @@ type TemplateData struct {
 
 	// BootstrapIP is address of the bootstrap node
 	BootstrapIP string
+
+	// Platform is the underlying provider the cluster is run on.
+	Platform            string
+	EtcdHeartBeat       string
```
Comment (hexfusion) [inline pkg/cmd/render/render.go:164]: nit EtcdHeartbeat

File: pkg/cmd/render/render_test.go [hunk 3]
```diff
@@ -97,12 +98,28 @@ spec:
...
+	infraConfig = `
+apiVersion: config.openshift.io/v1
+kind: Infrastructure
+metadata:
+  name: cluster
+spec:
+  cloudConfig:
+    name: ""
+status:
+  platform: AWS
```
Comment (hexfusion) [inline pkg/cmd/render/render_test.go:111]: can we have a test case where platform is empty?

Comment (alaypatel07) [reply pkg/cmd/render/render_test.go:111]: added it in the Platform Unit test. PTAL

**Pull request merged**